src/code/z_eff_blure.c

```diff
@@ -66,40 +68,977 @@ void func_800A81F0(EffectBlure* this, Vec3f* p1, Vec3f* p2) {
+        this->flags = initParams->flags;
+        this->drawMode = initParams->drawMode;
+        this->addAngleChange = initParams->addAngleChange;
+        this->addAngle = 0;
+        this->mode4Param = initParams->mode4Param;
+        this->altPrimColor = initParams->altPrimColor;
+        this->altEnvColor = initParams->altEnvColor;
+    }
+}
+
+void EffectBlure_Destroy(void* thisx) {
+}
+
+s32 EffectBlure_Update(void* thisx) {
+    EffectBlure* this = (EffectBlure*)thisx;
+    s32 i;
+
+    if (this == NULL) {
+        return 0;
+    }
+
+    if (this->numElements == 0) {
+        return 0;
+    }
+
+    while (true) {
+        if (this->elements[0].state == 0) {
+            for (i = 0; i < 15; i++) {
+                this->elements[i] = this->elements[i + 1];
+            }
+
+            this->elements[i].state = 2;
+            this->elements[i].p1.x = 0;
+            this->elements[i].p1.y = 0;
+            this->elements[i].p1.z = 0;
+            this->elements[i].p2.x = 0;
+            this->elements[i].p2.y = 0;
+            this->elements[i].p2.z = 0;
+            this->elements[i].flags = 0;
+            this->elements[i].timer = 0;
+
+            this->numElements--;
+            if (this->numElements <= 0) {
+                this->numElements = 0;
+                return 0;
+            }
+        } else {
+            break;
+        }
+    }
+
+    if (this->elements[0].state == 2) {
+        return 0;
+    }
+
+    for (i = 0; i < this->numElements; i++) {
+        this->elements[i].timer++;
+    }
+
+    if (this->elemDuration < this->elements[0].timer) {
+        for (i = 0; i < 15; i++) {
+            this->elements[i] = this->elements[i + 1];
+            if (this->elements[i].timer >= this->elemDuration) {
+                this->elements[i].timer = this->elemDuration;
+            }
+        }
+
+        this->elements[i].state = 2;
+        this->elements[i].p1.x = 0;
+        this->elements[i].p1.y = 0;
+        this->elements[i].p1.z = 0;
+        this->elements[i].p2.x = 0;
+        this->elements[i].p2.y = 0;
+        this->elements[i].p2.z = 0;
+        this->elements[i].flags = 0;
+        this->elements[i].timer = 0;
+
+        this->numElements--;
+        if (this->numElements <= 0) {
+            this->numElements = 0;
+            return 0;
+        }
+        return 0;
+    }
+
+    for (i = 0; i < this->numElements; i++) {
+        if (this->elements[i].timer >= this->elemDuration) {
+            this->elements[i].timer = this->elemDuration;
+        }
+    }
+
+    this->addAngle += this->addAngleChange;
+    return 0;
+}
+
+void EffectBlure_UpdateFlags(EffectBlureElement* elem) {
+    Vec3f sp64;
+    Vec3f sp58;
+    Vec3f sp4C;
+    Vec3f sp40;
+    EffectBlureElement* prev = elem - 1;
+    EffectBlureElement* next = elem + 1;
+    f32 sp34;
+    f32 sp30;
+    f32 sp2C;
+
+    if (((elem - 1)->state == 0) || ((elem + 1)->state == 0)) {
+        elem->flags &= ~3;
```

Comment:
```suggestion
        elem->flags &= ~(2 | 1);
```
My personal preference for bitwise flags is to write them out explicitly so that it's easier to see which bits are turned off, plus it'll be easier to document flag enums down the road (applies to a few places)

Reply:
Is this the preference in MM? In OoT we've generally avoided doing this, because all these flags would eventually have defines anyway, and it's slightly easier to automate splitting everything into defines if they are kept as the combined value.
But if MM is doing things differently, I don't mind making the change here

Reply:
I would say there isn't an MM preference, this has been more of a personal preference. Does it make it easier for scripts to split? I've always found it easier to read and document with it split, but I'm open to changing it if it's more convenient for tools